src/test/java/com/puppycrawl/tools/checkstyle/DefaultLoggerTest.java

```diff
@@ -116,6 +118,33 @@ public void testNewCtorWithTwoParameters() {
                 infoStream.toString().contains("java.lang.IllegalStateException: upsss"));
     }
 
+    @Test
+    public void testNullInfoStreamOptions() {
+        try {
+            final DefaultLogger logger = new DefaultLogger(new ByteArrayOutputStream(), null);
+            assertNotNull("Null instance", logger);
```

Comment:
isn't this assertion unnecessary since eventually we will hit `fail` below? Also when can a new instance return `null`?

Reply:
Without this assertion, the warning "The allocated object is never used" appears. See
https://travis-ci.org/checkstyle/checkstyle/jobs/378234050#L761
We need to use the logger somehow to make the eclipse compiler happy.

Reply:
happiness for all - is tough.
This nuance nobody will remember, it far from obvious. 
What if we will put in assert message "Null instance; assert required to calm down eclipse's 'The allocated object is never used' violation" or just make single line comment above assert ?

Reply:
>What if we will put in assert message "Null instance; assert required to calm down eclipse's 'The allocated object is never used' violation" or just make single line comment above assert ?

done

Reply:
This sounds familiar which we had to do with other tests.